src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -383,136 +360,53 @@ def handlePileup(self):
-        second pileup files of type "data".
-
-        """
-        mixModules, dataMixModules = [], []
-        prodsAndFilters = {}
-        prodsAndFilters.update(self.process.producers)
-        prodsAndFilters.update(self.process.filters)
-        for key, value in prodsAndFilters.items():
-            if value.type_() in ["MixingModule", "DataMixingModule", "PreMixingModule"]:
-                mixModules.append(value)
-            if value.type_() == "DataMixingModule":
-                dataMixModules.append(value)
-        return mixModules, dataMixModules
-
-    def _getPileupConfigFromJson(self):
-        """
-        There has been stored pileup configuration stored in a JSON file
-        as a result of DBS querrying when running PileupFetcher,
-        this method loads this configuration from sandbox and returns it
-        as dictionary.
-
-        The PileupFetcher was called by WorkQueue which creates job's sandbox
-        and sandbox gets migrated to the worker node.
-
-        """
-        workingDir = self.stepSpace.location
-        jsonPileupConfig = os.path.join(workingDir, "pileupconf.json")
-        self.logger.info("Pileup JSON configuration file: '%s'", jsonPileupConfig)
+        # Load pileup json 
         try:
             with open(jsonPileupConfig) as jdata:
                 pileupDict = json.load(jdata)
         except IOError:
             m = "Could not read pileup JSON configuration file: '%s'" % jsonPileupConfig
             raise RuntimeError(m)
-        return pileupDict
+
+        # Create a json with a list of files and events available 
+        # after dealing with PhEDEx/AAA logic
+        newPileupDict = {}
+        fileList = []
+        eventsAvailable = 0
+        for pileupType in self.step.data.pileup.listSections_():
+            useAAA = True if getattr(self.jobBag, 'trustPUSitelists', False) else False
+            self.logger.info("Pileup set to read data remotely: %s", useAAA)
+            for blockName in sorted(pileupDict[pileupType].keys()):
+                blockDict = pileupDict[pileupType][blockName]
+                if PhEDExNodeName in blockDict["PhEDExNodeNames"] or useAAA:
+                    eventsAvailable += int(blockDict.get('NumberOfEvents', 0))
+                    for fileLFN in blockDict["FileList"]:
+                        fileList.append(str(fileLFN))
+            newPileupDict[pileupType] = {"eventsAvailable": eventsAvailable, "FileList": fileList}
+        newJsonPileupConfig = os.path.join(self.stepSpace.location, "CMSSWPileupConfig.json")
+        self.logger.info("Generating json for CMSSW pileup script")
+        try:
+            with open(newJsonPileupConfig, 'wb') as f:
+                json.dump(newPileupDict, f)
+        except Exception as ex:
+            self.logger.exception("Error writing out process filelist json:")
+            raise
+
+        procScript = "cmssw_handle_pileup.py"
+        cmd = "%s --input_pkl %s --output_pkl %s --pileup_dict %s" % (
```

Comment:
I was going to suggest to skip this call if we do not have any files to be added as secondary files. However, it would break the current logic which is: always overwrite the list of secondary files.

With that said, do we need to provide a fix to the cmssw-tools for those cases we are providing an empty dictionary as input?

Reply:
I guess the answer is no, nothing else is needed, right?

Reply:
Correct, nothing else should be needed.